File: src/endstone_primebds/handlers/items.py
from endstone.event import PlayerPickupItemEvent

from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from endstone_primebds.primebds import PrimeBDS

def handle_item_pickup_event(self: "PrimeBDS", ev: PlayerPickupItemEvent):
    if not self.gamerules.get("can_pickup_items", 1):
        ev.is_cancelled = True
    return

File: src/handlers/items.cpp
/// @file items.cpp
/// Item pickup event handler.

#include "primebds/handlers/items.h"
#include "primebds/plugin.h"

namespace primebds::handlers {

    void handleItemPickupEvent(PrimeBDS &plugin, endstone::PlayerPickupItemEvent &event) {
        auto it = plugin.gamerules.find("can_pickup_items");
        if (it != plugin.gamerules.end() && !it->second)
            event.setCancelled(true);
    }

} // namespace primebds::handlers
